installinstallmacos.py

```diff
@@ -30,42 +30,113 @@
 import plistlib
 import subprocess
 import sys
-import time
 import urlparse
 import xattr
 from xml.dom import minidom
 from xml.parsers.expat import ExpatError
+from distutils.version import LooseVersion
 
 
-DEFAULT_SUCATALOG = (
-    'https://swscan.apple.com/content/catalogs/others/'
-    'index-10.13seed-10.13-10.12-10.11-10.10-10.9'
-    '-mountainlion-lion-snowleopard-leopard.merged-1.sucatalog.gz')
+DEFAULT_SUCATALOGS = {
+    '17': 'https://swscan.apple.com/content/catalogs/others/'
+          'index-10.13-10.12-10.11-10.10-10.9'
+          '-mountainlion-lion-snowleopard-leopard.merged-1.sucatalog',
+    '18': 'https://swscan.apple.com/content/catalogs/others/'
+          'index-10.14-10.13-10.12-10.11-10.10-10.9'
+          '-mountainlion-lion-snowleopard-leopard.merged-1.sucatalog',
+}
+
 
 SEED_CATALOGS_PLIST = (
     '/System/Library/PrivateFrameworks/Seeding.framework/Versions/Current/'
     'Resources/SeedCatalogs.plist'
 )
 
 
+def get_board_id():
+    '''Gets the local system board ID'''
+    ioreg_cmd = ['/usr/sbin/ioreg', '-p', 'IODeviceTree', '-r', '-n', '/', '-d', '1']
+    try:
+        ioreg_output = subprocess.check_output(ioreg_cmd).splitlines()
+        for line in ioreg_output:
+            if 'board-id' in line:
+                board_id = line.split(" ")[-1]
+                board_id = board_id[board_id.find('<"')+2:board_id.find('">')]
+                return board_id
+    except subprocess.CalledProcessError, err:
+        print "Board ID could not be identified. Error was: %s" % err
+        return ''
+
+
+def is_a_vm():
+    '''Determines if the script is being run in a virtual machine'''
+    sysctl_cmd = ['/usr/sbin/sysctl', 'machdep.cpu.features']
+    try:
+        sysctl_output = subprocess.check_output(sysctl_cmd)
+        cpu_features = sysctl_output.split(" ")
+        is_vm = False
```

Comment:
Move this before the try block so that if you do get an exception the function will return False instead of throwing another exception.

Reply:
Done